### PR TITLE
WP-2  warranty registration

### PR DESCRIPTION
**Pull request by arunRecbrT**
Do run migrations and RolePermissionSeeder to  create tables and seed database with new permissions

### CODE REVIEW

File: app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Index.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        return view('livewire.equipment.warranty.brand-configurator.index');
+    }
+
+    public function getConfiguredProperty()
+    {
+        if(!auth()->user()->can('equipment.warranty.view') ) return false;
+        return true;
+    }
+
+    public function create(){
```
Comment (anaskm) [inline app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Index.php:41]: @arunRecbrT please follow PSR-2 https://www.php-fig.org/psr/psr-2/

File: app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Index.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        'pageUpdated' => 'pageUpdated'
+    ];
+
+    public function render()
+    {
+        $this->authorize('view', BrandWarranty::class);
+        return view('livewire.equipment.warranty.brand-configurator.index');
+    }
+
+    public function getConfiguredProperty()
```
Comment (anaskm) [inline app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Index.php:35]: @arunRecbrT are we using this property anywhere?

File: app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Traits/FormRequest.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * Create new account
+     */
+    public function store()
+    {
+        $this->authorize('store', BrandWarranty::class);
+        $this->warranty->save();
+
+        session()->flash('success', 'Record created!');
```
Comment (anaskm) [inline app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Traits/FormRequest.php:91]: @arunRecbrT use livewire alert instead

File: app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Traits/FormRequest.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    public function brandUpdated($name, $value, $recheckValidation = false)
+    {
+        $this->fieldUpdated($name, $value, $recheckValidation);
+        $this->linesDisabled=false;
+        $this->lines = Line::where('brand_id',$value)->pluck('name', 'id');
+    }
+    public function linesUpdated($name, $value, $recheckValidation = false)
+    {
+        $this->fieldUpdated($name, $value, $recheckValidation);
+    }
```
Comment (anaskm) [inline app/Http/Livewire/Equipment/Warranty/BrandConfigurator/Traits/FormRequest.php:67]: @arunRecbrT no need of custom listener right?

**Comment by arunRecbrT** [comment]
index

brand configurator - Index page 
show
Show Page
 
create
edit
Create & Edit Pages

**Pull request merged**